### ISSUE TITLE
Implement `MongoDB` `ConfigStore` in Node.js SDK

### ISSUE THREAD

**Issue by rannn505** [configu/configu#146]
**Description**

- Plan a `Config` accommodation in the MongoDB storage.
- Add `MongoDBConfigStore` class to the Node.js SDK stores.
- We can use the native [mongodb](https://www.npmjs.com/package/mongodb) driver library during implementation.
- Or we can leverage the [ORMConfigStore](https://github.com/configu/configu/blob/main/ts/packages/node/src/stores/ORM.ts) implemented with `typeorm` that add a [mongodb support](https://typeorm.io/data-source-options#mongodb-data-source-options).

**Motivation**

Extending Configu's [available stores](https://configu.com/docs/config-store/#available-stores) collection.
This ConfigStore will be made available in the Configu CLI since the Node.js SDK is used internally.

**Additional context**

- https://www.mongodb.com/
- https://configu.com/docs/config-store/
- https://github.com/configu/configu/blob/main/types/ConfigStore.ts
- https://github.com/configu/configu/tree/main/ts/packages/node/src/stores

![image](https://github.com/configu/configu/assets/11806915/e603c3c3-703c-46d2-91a6-3c308089a8ce)

**Comment by Rohxn16** [comment]
Imma take look here

**Comment by SanjanaSogimatt** [comment]
Is the issue still open? Can I work on it?

**Comment by rannn505** [comment]
@Rohxn16 , @SanjanaSogimatt still open, feel free to go for it <3

**Comment by Tushar-4781** [comment]
Hi @rannn505 i want to give this issue a shot. Just wanted to confirm my understanding
1. we need to Create a class called MongoDBConfigStore that extends the base ConfigStore class and 
2. implement the get and set methods to interact with MongoDB for retrieving and storing key-value pairs.

**Comment by rannn505** [comment]
@Tushar-4781 Indeed, you can spin up Configu with other DB based ConfigStore and see the results.

**Comment by rannn505** [comment]
Hi @Tushar-4781,

You've nailed the understanding perfectly! Your task is indeed to create a MongoDBConfigStore class that extends our base ConfigStore class and to implement the get and set methods for MongoDB interactions.

Feel free to draw inspiration from the other DB stores within the Node SDK. Additionally, you have the freedom to opt for either the native MongoDB driver or the TypeORM based store - choose whichever aligns best with your implementation strategy.

We’re here to assist you throughout your contribution journey. Eagerly awaiting to review your work!

**Comment by rannn505** [comment]
@Tushar-4781 @Rohxn16 @SanjanaSogimatt just wanna to make sure you don't work on it individually, you are more than welcome to join forces in Discord and donate the pr together or help realize other stores.

**Comment by SanjanaSogimatt** [comment]
[configu-mongo.txt](https://github.com/configu/configu/files/12795578/configu-mongo.txt)
Okay so I'm a complete beginner so this is what I came up with do suggest me improvements with respect to anything

**Comment by RichardAkman** [comment]
Thanks for sharing your input, @SanjanaSogimatt. It's great to see new contributors like you getting involved! I wanted to let you know that @Sadaf-A  has already opened a PR and is actively addressing this issue. If you have any further questions or ideas, please don't hesitate to ask. Your involvement and ideas are always welcome!

**Comment by RichardAkman** [comment]
Hi @Tushar-4781 @Rohxn16 @SanjanaSogimatt, the issue is now available due to Sadaf being unavailable to continue work on his PR. Let me know which one of you wants to take it up or even work on it together!

**Comment by RichardAkman** [comment]
All yours @Nupoor10

**Comment by RichardAkman** [comment]
Hi @Nupoor10,

Since we haven’t seen a pull request since this issue was assigned and Hacktoberfest is underway, we will make this issue available to other contributors. Thank you for your interest in contributing!
